src/plot.c

```diff
@@ -96,3 +100,21 @@ void plot_dataframe_save(sc_dataframe * df)
 
 
 }
+
+/**
+ * @brief Frees the memory used within the dataframe
+ * @param df The dataframe to free
+ * @returns
+ */
+void plot_dataframe_free(sc_dataframe * df)
+{
+    /* Free all the df slices */
+    int i;
+    for( i=0; i< df->slice_no; i++)
+    {
```

Comment:
In kernel style C if there's just one line in the loop then you don't need the braces.